.changeset/rich-dolls-compete.md

```diff
@@ -0,0 +1,34 @@
+---
+"astro": minor
+---
+
+Improves the developer experience of the `500.astro` file by passing it a new `error` prop.
+
+When an error is thrown, the special `src/pages/500.astro` page now automatically receives the error as a prop. This allows you to display more specific information about the error on a custom 500 page.
```

Comment:
Should we document that the `error` prop only works when dynamically rendering `500.astro`? Since we can't get an error after it's prerendered.

Also, perhaps docs-specific but we should also caution about printing the error entirely as stuff like `.stack` may expose the server implementation.

Reply:
True, I'll update both changeset and docs